lib/compress.js

```diff
@@ -1053,7 +1053,7 @@ merge(Compressor.prototype, {
             var val = {};
             for (var i = this.properties.length; --i >= 0;) {
                 var prop = this.properties[i];
-                if (prop instanceof AST_ObjectKeyVal) {
+                if (typeof prop.key === "string") {
```

Comment:
Within this loop I'm constructing an object which I can evaluate the values of, filtering out all the others.

The idea is to keep the evaluation process forward, and then at the end see if we hit anything constant. If we don't (or there isn't a key match due to its leaf fails to evaluate), we just bail out.

Now that I'm thinking out loud, I think try-catch-ing the `ev(prop.value, compressor);` here would make sense as I want to skip over those failures.